File: 3169. Count Days Without Meetings.java
class Solution {
    public int countDays(int days, int[][] meetings) {
        Arrays.sort(meetings, (a,b) -> a[0] - b[0]);
        int i = 0;
        int answer = meetings[0][0] - 1;
        int end = 0;
        while (i < meetings.length) {
            end = meetings[i][1];
            while (i < meetings.length && meetings[i][0] <= end) {
                end = Math.max(end, meetings[i][1]);
                i++;
            }
            if (i < meetings.length) {
                answer += meetings[i][0] - end - 1;
            }
        }
        answer += days - end;
        return answer;
    }
}

File: 3169. Count Days Without Meetings.go
func countDays(days int, meetings [][]int) int {
	sort.Slice(meetings, func(i, j int) bool {
		return meetings[i][0] < meetings[j][0]
	})
	i, end, answer := 0, 0, meetings[0][0]-1
	for i < len(meetings) {
		end = meetings[i][1]
		for i < len(meetings) && meetings[i][0] <= end {
			end = max(end, meetings[i][1])
			i++
		}
		if i < len(meetings) {
			answer += meetings[i][0] - end - 1
		}
	}
	answer += days - end
	return answer
}